### PR TITLE
Add `cupyx.scipy.fft.next_fast_len`

### PR DESCRIPTION
**Pull request by asi1024**
Ready for #3560.
cc: @grlee77 and @leofang

### CODE REVIEW

**Comment by leofang** [comment]
Wow How much does caching help?

**Comment by leofang** [comment]
> Wow How much does caching help?

It really helps...Very cool 👍
```python
In [3]: %timeit nfl_cython(1000000)                                                                                                                                    
178 ns ± 0.81 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [4]: %timeit next_fast_len(1000000)  # this PR                                                                                                                        
271 ns ± 2.48 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

**Comment by leofang** [comment]
@asi1024 Could you take a look? I'm comparing the outputs between the two approaches.

```python
In [10]: not_match = 0 
    ...: for i in range(100000): 
    ...:     if nfl_cython(i) != next_fast_len(i): 
    ...:         not_match += 1 
    ...:          
    ...: print(not_match)                                                                                                                                              
57872
```

**Comment by asi1024** [comment]
> @asi1024 Could you take a look? I'm comparing the outputs between the two approaches.
I couldn't reproduce it... Could you tell me the smallest `i` does not match the return values?

**Comment by leofang** [comment]
I think in SciPy's approach, the preference is 2 > 3 > 5 > 7  if possible, whereas in your approach, the preference is to a length as small as possible:
```python
In [14]: nfl_cython(13)                                                                                                                                                
Out[14]: 15

In [15]: next_fast_len(13)                                                                                                                                             
Out[15]: 14

In [18]: nfl_cython(21)                                                                                                                                                
Out[18]: 24

In [19]: next_fast_len(21)                                                                                                                                             
Out[19]: 21
```
So my output is always larger than yours. While I think cuFFT doc does state a preference order (see [here](https://docs.nvidia.com/cuda/cufft/index.html#accuracy-and-performance)):
> In general the best performance occurs when using powers of 2, followed by powers of 3, then 5, 7.

It remains to be seen if it's actually true...

**Comment by leofang** [comment]
Sorry, scratch that 🤦🏻‍♂️ I messed up my local build....... 🙇 I rerun it and find no discrepancy. Apologize for the noise...

File: cupyx/scipy/fft/_helper.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    Returns:
+        int: The smallest fast length greater than or equal to the input value.
+
+    .. seealso:: :func:`scipy.fft.next_fast_len`
+
+    .. note::
+        It may return a different value to :func:`scipy.fft.next_fast_len`
+        as pocketfft's prime factors are different from cuFFT's factors.
+        For details, see
+        https://docs.nvidia.com/cuda/cufft/index.html#accuracy-and-performance
```
Comment (leofang) [inline cupyx/scipy/fft/_helper.py:41]: Just a nitpick to make the doc look nicer
```suggestion
        For details, see the `cuFFT documentation`_.
        
    .. _cuFFT documentation:
        https://docs.nvidia.com/cuda/cufft/index.html#accuracy-and-performance
```

File: docs/source/reference/scipy_fft.rst [hunk 2]
```diff
@@ -44,3 +44,13 @@ Code compatibility features
 
 .. note::
     To use ``scipy.fft.set_backend()`` together with an explicit ``plan`` argument requires SciPy version 1.5.0 or newer.
+
+
+Helper functions for FFT
+------------------------
```
Comment (leofang) [inline docs/source/reference/scipy_fft.rst:50]: Could you move this section to before `Code compatibility feature`?

**Comment by asi1024** [comment]
Fixed!

**Comment by grlee77** [comment]
Looks good! Thanks @asi1024 and @leofang

**Comment by leofang** [comment]
Jenkins, test this please

**Pull request merged**